src/Autocomplete/src/Form/AutocompleteChoiceTypeExtension.php

```diff
@@ -47,6 +47,8 @@ public function finishView(FormView $view, FormInterface $form, array $options)
 
         $attr = $view->vars['attr'] ?? [];
 
+        $attr['required'] = $view->vars['required'];
```

Comment:
do we only want to do this for `ChoiceType` @weaverryan ?

Reply:
I think so yes. However, the `required` attribute is removed from the `select` for a reason in https://github.com/symfony/symfony/blob/b1b77f366d42a58c1cd0879c605b997d05279d01/src/Symfony/Bridge/Twig/Resources/views/Form/form_div_layout.html.twig#L59-L61 ... though I don't know what that reason is :). I'm wondering if we should avoid re-adding that attribute, and instead pass in a Stimulus variable.

Reply:
Ping on this! I think this is the only thing left - I'm thinking we should choose a variable other than `required`.

Reply:
@weaverryan my concern with this is that this will not work with bare (non symfony) forms, i.e. https://symfony.com/bundles/ux-autocomplete/current/index.html#manually-using-the-stimulus-controller

thoughts?

Reply:
@weaverryan see pending question above :)

Reply:
I did some digging. The original behavior in https://github.com/symfony/symfony/blob/b1b77f366d42a58c1cd0879c605b997d05279d01/src/Symfony/Bridge/Twig/Resources/views/Form/form_div_layout.html.twig#L59-L61 comes from https://github.com/symfony/symfony/issues/8942 - it's an HTML5 spec thing. So I'd like to not add `required` (or, in the case of people manually creating `select` elements - force them to add `required`) in invalid situations.

Playing with the demo for this plugin - https://tom-select.js.org/plugins/clear-button/#clear-button - it feels like the invalid situation is when you have a `single` select and NO placeholder (defined as the first `option` having a `value=""`). If you DO have a placeholder, the "x" correctly resets things back to that value. But if you don't, that's when things get weird and it changes your select to an empty value... even though none originally existed.

I think this is what we should trigger on:

```
const isMultiple = !this.selectElement || this.selectElement.multiple;
const isSelectWithPlaceholder = this.selectElement && this.selectElement.options[0] && this.selectElement.options[0].value === '' ;
if (!this.formElement.disabled && !isMultiple && !isSelectWithPlaceholder) {
```

Reply:
@weaverryan just pushed this JS changed, but noticed it does not take `this.formElement.required` into account at all now.
Was this your intention?
long time since i looked at this so i'm a bit hazy...

Reply:
i think we still need to take `required` into account? added this. please take a look.

Reply:
i don't think we need to do this? text type can be required or not, too?